### PR TITLE
Add ChatDomain migration guide

### LINKED ISSUE #3034: Write OfflinePlugin documentation
### Context & proposal

Write documentation for `OfflinePlugin`. Make sure to cover:
- [ ] What's the architecture
- [ ] How to initialize the `OfflinePlugin`
- [ ] How to make calls
- [ ] How to get state

Make sure that existing documentation is migrated (both in the CMS and Docusaurus) and doesn't use `ChatDomain` approach

### PR DESCRIPTION
**Pull request by adasiewiczr**
closes #3034 

### 🎯 Goal

Add `ChatDomain` migration guide.

### 🛠 Implementation details

1. Add migration guide.
2. Unify `addMembers` and `removeMembers` API

### 🧪 Testing
**NOTE: Some of the link won't work correctly before the release **

1. Checkout the branch
3. Run `npx stream-chat-docusaurus -i -s`
4. Review the migration guide page

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [x] Release notes and docs clearly describe changes

### CODE REVIEW

**Review by MarinTolic — APPROVED** [review]
I think this is a good migration guide explaining the most important aspect of having a single point of access for customers.

**Review by leandroBorgesFerreira — APPROVED** [review]

File: docusaurus/docs/Android/02-client/06-guides/07-chatdomain-migration.mdx [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# ChatDomain Migration
+
+**v5.0.0** release brings a big change to the offline support library - it replaces `ChatDomain` with a new, easy-to-use `OfflinePlugin`.
+You can read more about the motivation behind the effort and featured changes in the [announcement blog post](https://getstream.io/blog/).  <!--TODO: Replace with a real link -->
```
Comment (skydoves) [inline docusaurus/docs/Android/02-client/06-guides/07-chatdomain-migration.mdx:4]: The permalink might be `https://getstream.io/blog/stream-chat-android-v5-sdk-release/`. I will keep you updated on this.

**Review by skydoves — APPROVED** [review]
Looks great to me!

File: docusaurus/docs/Android/02-client/06-guides/07-chatdomain-migration.mdx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+:::note
+`chatClient.channel(cid)` returns a `ChannelClient` that uses `ChatClient` under the hood and simplifies performing actions in particular channel.
+:::
+
+The approach mentioned above can be applied to all API-call-related `ChatDomain` methods.
+
+### Observing the State
+
+We've renamed the objects used to obtain a state:
```
Comment (filbabic) [inline docusaurus/docs/Android/02-client/06-guides/07-chatdomain-migration.mdx:69]: `obtain state`

**Review by filbabic — APPROVED** [review]
Looks great! 

Just a simple comment ^^

**Pull request merged**